### ISSUE TITLE
resolveSecondary command working only if it is on the first line of the macro

### ISSUE THREAD

**Issue by rftw** [UltimateHackingKeyboard/firmware#545]
I am using the latest firmware release (9.1.3#55f4531) and agent version 2.01. As the title states, resolveSecondary command is working only when it is on the first line of the macro. If it is preceded by any other command or comment, it's not working.

working:
```
resolveSecondary 350 @1 @3
holdKey j
break
holdKey RC- 
```

Not working:
```
// Dual role
resolveSecondary 350 @1 @3
holdKey j
break
holdKey RC- 
```
Not working:
```
ifAnyMod final holdKey j
resolveSecondary 350 @1 @3
holdKey j
break
holdKey RC- 
```

**Comment by kareltucek** [comment]
Nice catch! Should be fixed by #548.

@mondalaci looks like you are not tagging the releases? If the tars are to be identified by tags preferentially, the tag needs to be known during build, which means that retroactive tagging via github is not sufficient...

**Comment by mondalaci** [comment]
I must have added the `v9.1.3` tag after creating the uploaded release, which unintentionally resulted in the current `55f4531` sha being used as the tag field in the package.json of the release. This makes Agent show the redundant and perhaps confusing sha for the 9.1.3 release. Still, Agent should be able to fetch the smart macro documentation when it's unavailable based on the sha, so it fulfills its purpose.

Do you see any other issues regarding the tag field?

For the sake of clarity, I think make-release.js should error out when encountering a tag that doesn't equal to `v${firmwareVersion}`, and allow for the `--allowSha` parameter to intentionally create a non-tagged release. Can you send us a PR for it?

Please create new issues in the future instead of commenting in existing ones. I think this is an unrelated issue.
